https://www.jianshu.com/p/a0a87ba6ef12

 源代码更改， 原来那么写报错


![1573780573707](踩坑记录.assets/1573780573707.png)

1.

错误：

```python
>>> import mujoco_py                                                                             
    raise error.MujocoDependencyError('To use MuJoCo, you need to either populate ~/.mujoco/mjkey.txt and ~/.mujoco/mjpro131, or set the MUJOCO_PY_MJKEY_PATH and MUJOCO_PY_MJPRO_PATH environment variables appropriately. Follow the instructions on https://github.com/openai/mujoco-py for where to obtain these.')                      
mujoco_py.error.MujocoDependencyError: To use MuJoCo, you need to either populate ~/.mujoco/mjkey.txt and ~/.mujoco/mjpro131, or set the MUJOCO_PY_MJKEY_PATH and MUJOCO_PY_MJPRO_PATH environment variables appropriately. Follow the instructions on https://github.com/openai/mujoco-py for where to obtain these.                                                                                                   
```

解决

![1573780214418](踩坑记录.assets/1573780214418.png)



2.

```ruby
Traceback (most recent call last):                                                                            
  File "<stdin>", line 1, in <module>                                                                         
  File "C:\anaconda3\envs\cs294hw1\lib\site-packages\mujoco_py\__init__.py", line 4, in <module>              
    from .mjviewer import MjViewer                                                                            
  File "C:\anaconda3\envs\cs294hw1\lib\site-packages\mujoco_py\mjviewer.py", line 7, in <module>              
    from . import mjcore, mjconstants, glfw                                                                   
  File "C:\anaconda3\envs\cs294hw1\lib\site-packages\mujoco_py\mjcore.py", line 6, in <module>                
    from .mjlib import mjlib                                                                                  
  File "C:\anaconda3\envs\cs294hw1\lib\site-packages\mujoco_py\mjlib.py", line 21, in <module>                
    mjlib = cdll.LoadLibrary(os.path.abspath(libfile))                                                        
  File "C:\anaconda3\envs\cs294hw1\lib\ctypes\__init__.py", line 426, in LoadLibrary                          
    return self._dlltype(name)                                                                                
  File "C:\anaconda3\envs\cs294hw1\lib\ctypes\__init__.py", line 348, in __init__                             
    self._handle = _dlopen(self._name, mode)                                                                  
OSError: [WinError 193] %1 不是有效的 Win32 应用程序。   
```

https://stackoverflow.com/questions/38766267/python-binding-for-mujoco-physics-library-using-mujoco-py-package

![1573780324903](踩坑记录.assets/1573780324903.png)





3.

![1573720092059](踩坑记录.assets/1573720092059.png)

PATH=C:\Users\28172\Desktop\mujoco\mjpro131\bin



4.

![1573720511114](踩坑记录.assets/1573720511114.png)

![1573725832961](踩坑记录.assets/1573725832961.png)



5.

![1573726101720](踩坑记录.assets/1573726101720.png)

![1573726409894](踩坑记录.assets/1573726409894.png)



6.

我索性就不用pip install 'gym[mujoco]'

直接下github的官方gym ，手动下

下好后，调代码，把python2.7的一些东西，包的调用添加好

调好后出现以下问题

![1573727912304](踩坑记录.assets/1573727912304.png)

更改

![1573728478768](踩坑记录.assets/1573728478768.png)

7.

![1573729646665](踩坑记录.assets/1573729646665.png)

![1573728599693](踩坑记录.assets/1573728599693.png)

把版本改成0.7.4

8.

![1573730025157](踩坑记录.assets/1573730025157.png)

windows 命名限制

![1573730277392](踩坑记录.assets/1573730277392.png)

把 time 相关的去掉



9.

![1573731032158](踩坑记录.assets/1573731032158.png)

pip3 install lxml

不加3 还不行 （有点奇怪 ） 下了还是没得用

![1573731453372](踩坑记录.assets/1573731453372.png)

这么就好了



10.

![1573731500581](踩坑记录.assets/1573731500581.png)

改成 items



11.

吐了 又是这些问题

![1573731793908](踩坑记录.assets/1573731793908.png)

![1573745423462](踩坑记录.assets/1573745423462.png)



12.

Making new env: CentipedeSix-v1
w library wasn't compiled to use SSE instructions, but these are available on your machine and could speed up CPU computations.

```
在main函数加这些
import os
os.environ['TF_CPP_MIN_LOG_LEVEL']='2'
import tensorflow as tf
```

![1573745709158](踩坑记录.assets/1573745709158.png)

13.又是windows 命名错误

![1573748413741](踩坑记录.assets/1573748413741.png)

success

![1573748484623](踩坑记录.assets/1573748484623.png)

一番努力终于调通



不过用他那个train 又出错了

1.![1573786804045](踩坑记录.assets/1573786804045.png)

python3.6 移除了itervalues  凡是遇到 iter-  的时候都删掉 iter

2.

![1574238606344](踩坑记录.assets/1574238606344.png)

![1574238611612](踩坑记录.assets/1574238611612.png)

这个其实不是问题，是TensorFlow版本的原因，在TensorFlow1.7会报错，以前的版本无影响。

![1574239750237](踩坑记录.assets/1574239750237.png)

![1574239770565](踩坑记录.assets/1574239770565.png)

发现问题

![1574240009605](踩坑记录.assets/1574240009605.png)

3.

![1574240109436](踩坑记录.assets/1574240109436.png)

确实没法了

ValueError: Shape must be rank 2 but is rank 4 for 'MLP_embedding_node_type_root/layer_0/MatMul' (op: 'MatMul') with input shapes: [?,5], [0,1075052544,0,1077936128].



顶不住， 确实， 矩阵都不对， 我怀疑这篇论文了

---

![1574322212771](踩坑记录.assets/1574322212771.png)



---



想使用 render

报错

![1574163600890](踩坑记录.assets/1574163600890.png)

![1574163589288](踩坑记录.assets/1574163589288.png)

![1574322701459](踩坑记录.assets/1574322701459.png)